contracts/PartyBidFactory.sol

```diff
@@ -34,20 +34,38 @@ contract PartyBidFactory {
     address public immutable tokenVaultFactory;
     address public immutable weth;
 
+    //======== Mutable storage =========
+
+    // PartyBid proxy => block number deployed at
+    mapping(address => uint256) public deployedAt;
+
     //======== Constructor =========
 
     constructor(
         address _partyDAOMultisig,
         address _tokenVaultFactory,
-        address _weth
+        address _weth,
+        address _logicMarketWrapper,
+        address _logicNftContract,
+        uint256 _logicTokenId,
+        uint256 _logicAuctionId
     ) {
         partyDAOMultisig = _partyDAOMultisig;
         tokenVaultFactory = _tokenVaultFactory;
         weth = _weth;
         // deploy logic contract
-        logic = address(
-            new PartyBid(_partyDAOMultisig, _tokenVaultFactory, _weth)
+        PartyBid _logicContract = new PartyBid(_partyDAOMultisig, _tokenVaultFactory, _weth);
+        // initialize logic contract
+        _logicContract.initialize(
```

Comment:
do we need to send in real values here (and take them in the constructor)? or could we just hard-code in null addresses and zeros?

Reply:
we need to send in real values or else `initialize` will revert (because it does some checks on the inputs)